### PR TITLE
Azure-Pipelines: update *.yml file and build scripts

### PR DESCRIPTION
**Pull request by amiclaus**
With this PR, the Azure-Pipelines integration is done.
Build jobs:
- code style check
- static analysis
- libtinyiiod module build

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
-    echo See https://aka.ms/yaml
-  displayName: 'Run a multi-line script'
+  pool:
+      vmImage: $(imageName)
+  steps:
+  - checkout: self
+    condition: eq(variables.runCondition, true)
+    fetchDepth: 50
+    clean: true
+    persistCredentials: true
```
Comment (commodo) [inline azure-pipelines.yml]: is this pushing back anything?
if not, i would not leave the credentials stored;

**Review by commodo — APPROVED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,19 +1,42 @@
...
+- dev/*
+
+pr:
+- main
+- master
 
-pool:
-  vmImage: 'ubuntu-latest'
+variables:
+  isMain: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
```
Comment (commodo) [inline azure-pipelines.yml]: isMain doesn't look used;
runCondition seems always true; will it be used?
targetBranch/TARGET_BRANCH doesn't seem used ; will it be used?

**Comment by amiclaus** [comment]
V2:
- remove unused variable `isMain`
- persistCredentials: set to `false`

File: azure-pipelines.yml (hunk 2 above)
Comment (amiclaus) [reply azure-pipelines.yml]: done. removed the unused variable

**Pull request merged**